src/mvAppItem.cpp

```diff
@@ -3947,18 +4056,20 @@ DearPyGui::GetEntityParser(mvAppItemType type)
             MV_PARSER_ARG_SHOW)
         );
 
-        args.push_back({ mvPyDataType::DoubleList, "x" });
-        args.push_back({ mvPyDataType::DoubleList, "y" });
-        args.push_back({ mvPyDataType::DoubleList, "negative" });
-        args.push_back({ mvPyDataType::DoubleList, "positive" });
-        args.push_back({ mvPyDataType::Bool, "contribute_to_bounds", mvArgType::KEYWORD_ARG, "True" });
-        args.push_back({ mvPyDataType::Bool, "horizontal", mvArgType::KEYWORD_ARG, "False" });
-
-        setup.about = "Adds an error series to a plot.";
+        args.push_back({ mvPyDataType::DoubleList, "values" });
+        args.push_back({ mvPyDataType::StringList, "label_ids" });
+        args.push_back({ mvPyDataType::Integer, "item_count" });
```

Comment:
`item_count` and `group_count`: We already have the total length of the `values` list, and can therefore deduce one of these arguments from another. I suggest to replace them with a single value specifying the number of bars in a group, and name it `group_size` (see also comments on the current `group_size` argument below). Then then number of items can be calculated from the list size.

What's more, I think we need to add more safety around list sizes. Right now, if `labels` is shorter than the number of groups, DPG will segfault. Since both `values` and `labels` are positional args (i.e. both will for sure be passed in), we can check that their sizes are correct (e.g. `values` is a multiple of `group_size`, and `labels` is no longer than `len(values)/group_size`), and raise an exception if they aren't.

Reply:
Where do you suggest to put these checks? I was thinking in the `draw` function: maybe it's not the ideal place to put them, but otherwise I should put them in `set_positional_configuration`, `set_configuration` and `fill_configuration_dict` and I should replicate code and I don't love it tbh.

Reply:
The drawback of checking them in `draw_group_bar_series` is that we can't raise an exception there. I'd prefer to signal the error right away. This means we need to check these sizes in `set_positional_configuration` and `set_configuration`, but not in `fill_configuration_dict`. To prevent code duplication, let's add a helper function like `ValidateBarGroupConfig(mvGroupBarSeriesConfig& outConfig)` and call it from both `set_positional_configuration` and `set_configuration`. You can even declare it static so that it's not visible outside of mvPlotting.cpp.

To raise an exception, use `mvThrowPythonError`. If I remember correctly, there's no need to return anything to Python in this case. Don't look at how DPG does it - in many cases throughout DPG code, it returns `None` after `mvThrowPythonError` - this is actually wrong and it crashes in debug mode (documented in #2097).

Reply:
Okay, I was implementing this and then I recognized that if I use the function as you told me (with `mvGroupBarSeriesConfig& outConfig` as parameter) I can check the state only when it's already been set, then if there's some error I should restore back to how it was.
Or `mvThrowPythonError` takes care of it by itself? (But I don't think so)

Reply:
`mvThrowPythonError` will just store exception info inside of Python so that when it gets back from C++, it raises the exception.

Yes, looks like we need to either back up the old state, or to have a function like `SetBarGroupData(mvGroupBarSeriesConfig& outConfig, some-other-parms)` that would store `values`, `labels`, and `group_size` in the config in a single atomic operation (first checking that new values are valid). "Some-other-parms" here could be three `PyObject*` obtained from keyword args or from positional args, and `SetBarGroupData` would parse them into local variables, replacing NULLs with values from the config. It can then validate the values and store in the config.

Reply:
Now it _should_ be okay, I tested it and it seems to work

Reply:
Still has some issues...

1. If positional arguments are incorrect, `set_positional_configuration` will restore the default values after `ValidateBarGroupConfig`, and `outConfig.group_size` will be zero (that's the default in `mvBarGroupSeriesConfig`). Since `ValidateBarGroupConfig` does not check it for being zero, it crashes on the next call, which happens in `set_configuration`. That is, if parameters on the `add_bar_group_series` are incorrect, the application will crash with a divide-by-zero even before raising a Python exception.

Notice how `ValidateBarGroupConfig` gets called twice on item creation. While it would be good to check for `group_size != 0` in `ValidateBarGroupConfig` itself (and we should probably do that), it's not sufficient: otherwise the second call (from `set_configuration`) might override the error raised in the first call, and for any incorrect arguments the user will always be getting "group_size must not be zero".

For this "backup of config values" to work properly, we must guarantee that at any moment (other than when the values are backed up), _including the default config_, the values are valid, so that when restoring the backup, we're getting a valid configuration. That is, add an initializer `group_size = 1` in `mvBarGroupSeriesConfig`.

2. Currently, `group_size` is passed to the `group_count` argument of `ImPlot::PlotBarGroups`, and therefore defines the number of groups rather than the group size. We need to swap `item_count` and `config.group_size` in the call to `ImPlot::PlotBarGroups`.

3. A typo in one of `ValidateBarGroupConfig` messages: "equel" -> "equal".

Reply:
Please check next commit... I don't know if I completely got you, or better, I don't know if I resolved all the possible problems in this way.

Reply:
Unfortunately it still has some issues.

1. The "must not" text that [I mentioned earlier](https://github.com/hoffstadt/DearPyGui/pull/2275#discussion_r1659967821).
2. `label_ids.size()` must be validated against `group_size`, not `item_count`.
3. The value is not backed up properly because `outConfig.value` is a `shared_ptr`.

Here's a fix for all those issues:
https://github.com/v-ein/DearPyGui-fixes/compare/ce0a50705ae275edd94286fa9d89990c72f708fe...bugfix/bar-group-validation